### ISSUE TITLE
Dialog guidance page component demo contains button without SR announced label

### ISSUE THREAD

**Issue by DBD324** [mi6/ic-design-system#612]
## Summary of the bug

On the dialog guidance page, the first component demo contains a button visually titled 'Launch dialog'. 

When interacting with this button with a SR (NVDA), the label is not announced. The SR only announces 'clickable button'. 

The other buttons in the code component demos seem to work fine and announce their visual label. 

## 🪜 How to reproduce

Tell us the steps to reproduce the problem:

1. Go to page: Dialog component page
2. Click on: Guidance tab
3. Open NVDA/Voiceover
4. Tab to the component demo and the button within. 
5. Hear the error. 

## 📸 Screenshots or code

If you can, add screenshots to show the problem. If you need to show a code example, you can include a snippet or link to a repository with an example.

## 🖥 📱 Device

- Type: Desktop
- Device: Windows
- OS version:
- Browser version: Edge
- SR: NVDA

## 🧐 Expected behaviour

When navigating to the button with a SR using either the focus keys or the virtual cursor, then the button label should be announced by the screen reader. 

## 📝 Acceptance Criteria

If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil.

_Given_ a user is navigating the dialog guidance page with a SR
_When_ navigating to the 'launch dialog' button in the first component demo using either the tab keys or the virtual cursor
_Then_ the SR should announce the button label and not just 'Clickable button'

## Additional info

Tell us anything else useful to help us fix or understand the problem.

**Comment by DBD324** [comment]
- Tested with NVDA 2023.1 on Edge in Windows virtual machine. No buttons announce their labels. 
- Tested with NVDA 2023.2 on Edge in different Windows virtual machine. Button labels are announced fine. 
- Tested with Voiceover on Chrome in Macbook. Only 'launch dialog' button in component demo does not announce label. All other buttons announce label fine.

**Comment by GCHQ-Developer-299** [comment]
I think this might be a Gatsby issue - i've noticed that if you click the Launch Dialog button, click Confirm to close the dialog, then the button's label gets read out correct.  This behaviour also remedies a similar bug [where Table of Contents links don't read properly](https://github.com/mi6/ic-design-system/issues/699)
